src/index.ts

```diff
@@ -133,9 +148,21 @@ function createStreamingClientMethod(grpcClient: DynamicMethods, name: string) {
   return function(...args: any[]) {
     return new Observable(observer => {
       const call = grpcClient[name](...args);
-      call.on('data', (data: any) => observer.next(data));
-      call.on('error', (error: any) => observer.error(error));
-      call.on('end', () => observer.complete());
+
+      const onData = (data: any, _: any, cb:any) => {
+        observer.next(data)
+        cb();
+      };
+      const onError = (error: any) => observer.error(error);
+
+      const onEnd = (cb: any) => {
+        observer.complete()
+        cb();
+        call.removeListener('error', onError);
```

Comment:
Don't leak listeners and use through2 to not use legacy steaming api of `data` and `end` .